File: solutions/025/025.py
"""
Project Euler - Problem Solution 025
Copyright (c) Justin McGettigan. All rights reserved.
https://github.com/jwmcgettigan/project-euler-solutions
"""

def newTerms(f): return [f[1], f[0] + f[1]]
def index_of_1000_digit_fibonacci_value():
  # repurposed from problem 002
  index, f = 1, [1, 1]
  while(len(str(f[0])) < 1000):
    f = newTerms(f)
    index += 1
  return index

if __name__ == "__main__":
	print(index_of_1000_digit_fibonacci_value())

File: solutions/025/025.js
/**
 * Project Euler - Problem Solution 025
 * Copyright (c) Justin McGettigan. All rights reserved.
 * https://github.com/jwmcgettigan/project-euler-solutions
 */

const newTerms = (f) => [f[1], f[0] + f[1]];
const indexOf1000DigitFibonacciValue = () => {
  // repurposed from problem 002
  let index = 1, f = [BigInt(1), BigInt(1)];
  while(f[0].toString().length < 1000) {
    f = newTerms(f);
    index++;
  }
  return index;
};

console.log(indexOf1000DigitFibonacciValue());